litestar/contrib/jinja.py

```diff
@@ -29,15 +29,26 @@
 class JinjaTemplateEngine(TemplateEngineProtocol["JinjaTemplate"]):
     """The engine instance."""
 
-    def __init__(self, directory: DirectoryPath | list[DirectoryPath]) -> None:
+    def __init__(
+        self,
+        directory: DirectoryPath | list[DirectoryPath] | None = None,
+        engine_instance: Environment | None = None,
+    ) -> None:
         """Jinja2 based TemplateEngine.
 
         Args:
             directory: Direct path or list of directory paths from which to serve templates.
+            engine_instance: A jinja2 Environment instance.
         """
-        super().__init__(directory=directory)
-        loader = FileSystemLoader(searchpath=directory)
-        self.engine = Environment(loader=loader, autoescape=True)
+
+        super().__init__(directory, engine_instance)
+        if directory and engine_instance:
+            raise ImproperlyConfiguredException("You must provide either a directory or a jinja2 Environment instance.")
+        if directory:
+            loader = FileSystemLoader(searchpath=directory)
+            self.engine = Environment(loader=loader, autoescape=True)
+        elif engine_instance:
```

Comment:
Given the above logic, shouldn't we raise an `ImproperlyConfiguredException` if both path and environment are given? And the docs need to clarify this too.

Reply:
not sure I follow you here since https://github.com/litestar-org/litestar/pull/2195/files#diff-4fda45c329a2b972920a33ce02381a562f8df1f939810be863bbfffcd443715fR45-R46 is exactly that

Reply:
added a doc mention in https://github.com/litestar-org/litestar/pull/2195/commits/30ad82834471beed00f674b15bb4acc380820162